hoot-core/src/main/cpp/hoot/core/io/HootNetworkRequest.cpp

```diff
@@ -159,4 +198,22 @@ int HootNetworkRequest::_getHttpResponseCode(QNetworkReply* reply)
   return -1;
 }
 
+void HootNetworkRequest::_setOAuthHeader(QNetworkAccessManager::Operation http_op, QNetworkRequest& request)
+{
+  //  Convert the operation format
+  OAuth::Http::RequestType op;
+  switch (http_op)
+  {
+  case QNetworkAccessManager::GetOperation:   op = OAuth::Http::Get;    break;
+  case QNetworkAccessManager::PutOperation:   op = OAuth::Http::Put;    break;
+  case QNetworkAccessManager::PostOperation:  op = OAuth::Http::Post;   break;
+  default:                                                              return;
+  }
+  //  Create the client object and get the HTTP header
+  OAuth::Client requestClient(_consumer.get(), _tokenRequest.get());
+  string header = requestClient.getHttpHeader(op, request.url().toString().toStdString());
+  //  Set the Authorization header for OAuth
+  request.setRawHeader("Authorization", QString(header.c_str()).toUtf8());
+}
+
```

Comment:
The `Authorization` header is created using the keys/secrets that creates a signature to validate the request with OAuth.  Since the signature hash includes the HTTP operation, that is converted here.